### PR TITLE
unfork ed25519-dalek

### PR DESCRIPTION
**Pull request by rob-solana**
#### Problem
ed25519-dalek upstream has improved, PartialEq excuse for forking holds no water

 #### Summary of Changes
 unfork

Fixes #

### CODE REVIEW

File: ledger/Cargo.toml [hunk 1]
```diff
@@ -34,7 +34,7 @@ solana-measure = { path = "../measure", version = "0.21.0" }
 solana-merkle-tree = { path = "../merkle-tree", version = "0.21.0" }
 solana-metrics = { path = "../metrics", version = "0.21.0" }
 solana-perf = { path = "../perf", version = "0.21.0" }
-solana-ed25519-dalek = "0.2.0"
+ed25519-dalek = "0.9"
```
Comment (garious) [inline ledger/Cargo.toml]: Can we jump to 1.0.0-pre.2?

Comment (rob-solana) [reply ledger/Cargo.toml]: if you'd prefer

Comment (rob-solana) [reply ledger/Cargo.toml]: hm.  some libra issues:
```
cd ~/dev/solana &&  cargo test -p solana-sdk -- --nocapture
    Updating crates.io index
error: failed to select a version for `ed25519-dalek`.
    ... required by package `solana_libra_crypto v0.0.1-sol4`
    ... which is depended on by `solana_libra_config v0.0.1-sol4`
    ... which is depended on by `solana_libra_language_e2e_tests v0.0.1-sol4`
    ... which is depended on by `solana-librapay-api v0.21.0 (/home/rwalker/dev/solana/programs/librapay_api)`
    ... which is depended on by `solana-bench-tps v0.21.0 (/home/rwalker/dev/solana/bench-tps)`
    ... which is depended on by `solana-local-cluster v0.21.0 (/home/rwalker/dev/solana/local_cluster)`
versions that meet the requirements `= 1.0.0-pre.1` are: 1.0.0-pre.1

all possible versions conflict with previously selected packages.

  previously selected package `ed25519-dalek v1.0.0-pre.2`
    ... which is depended on by `solana-core v0.21.0 (/home/rwalker/dev/solana/core)`
    ... which is depended on by `solana-archiver v0.21.0 (/home/rwalker/dev/solana/archiver)`

failed to select a version for `ed25519-dalek` which could resolve this conflict
```

Comment (garious) [reply ledger/Cargo.toml]: I guess `1.0.0-pre.1` then...  @jackcmay, need non-protocol dependencies out of the main build!

**Comment by garious** [comment]
:shipit:

File: ledger/Cargo.toml (hunk 1 above)
Comment (jackcmay) [reply ledger/Cargo.toml]: @garious I think this is more related to bench-tps's dependency on both core and libra and depending on a non-semver supported version.  I doubt moving non-protocol dependencies out of the tree will help here.

**Pull request merged**